Tests/BackendIntegrationTests/BaseBackendIntegrationTests.swift

```diff
@@ -52,6 +52,10 @@ class BaseBackendIntegrationTests: XCTestCase {
             throw ErrorUtils.configurationError(message: "Must set configuration in `Constants.swift`")
         }
 
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+            await self.finishAllUnfinishedTransactions()
+        }
```

Comment:
let's add a comment explaining why this is needed, it'd be reasonable for a maintainer to assume this is a bug

Reply:
Done